File: Stage9/C/Problem9506.c
#include <stdio.h>

int main(void) {
    int n;

    while (1) {
        scanf("%d", &n);
        if (n == -1) break;

        int sum = 0;
        int divisors[1000];
        int count = 0;

        for (int i = 1; i <= n / 2; i++) {
            if (n % i == 0) {
                divisors[count++] = i;
                sum += i;
            }
        }
        
        if (sum == n) {
            printf("%d = ", n);
            for (int i = 0; i < count; i++) {
                if (i > 0) printf(" + ");
                printf("%d", divisors[i]);
            }
            printf("\n");
        } else {
            printf("%d is NOT perfect.\n", n);
        }
    }

    return 0;
}

File: Stage9/Python/Problem9506.py
def main():
    while (True):
        n = int(input())
        if n == -1:
            break
        
        divisors = [i for i in range(1, n // 2 + 1) if n % i == 0]
        if sum(divisors) == n:
            print(f"{n} = {' + '.join(map(str, divisors))}")
        else:
            print(f"{n} is NOT perfect.")
    
if __name__ == "__main__":
    main()